rollout/service.go

```diff
@@ -258,8 +257,16 @@ func (c *rolloutContext) ensureSVCTargets(svcName string, rs *appsv1.ReplicaSet)
 	if err != nil {
 		return err
 	}
-	if svc.Spec.Selector[v1alpha1.DefaultRolloutUniqueLabelKey] != rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey] {
-		err = c.switchServiceSelector(svc, rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey], c.rollout)
+	currSelector := svc.Spec.Selector[v1alpha1.DefaultRolloutUniqueLabelKey]
+	desiredSelector := rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey]
+	if currSelector != desiredSelector {
+		// ensure we have available replicas, otherwise we will point the service to nothing
+		if rs.Status.AvailableReplicas == 0 {
```

Comment:
If you only check against zero then you can have the situation where 1 pod is available but the service needs 100 causing an outage.

This should check that AvailableReplicas == DesiredReplicas.

Reply:
Good catch. Will check for full availability.